client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -129,6 +132,19 @@ export function getEnhancedTasks(
 						},
 					};
 					break;
+				case 'unlock_styles':
+					if ( ! isEnabled( 'limit-global-styles' ) || ! globalStylesInUse ) {
+						return;
+					}
+					taskData = {
+						title: translate( 'Activate advanced design customization tools' ),
+						isCompleted: ! shouldLimitGlobalStyles,
+						actionDispatch: () => {
+							recordTaskClickTracksEvent( flow, task.isCompleted, task.id );
+							window.location.replace( `/site-editor/${ siteSlug }?unlock-styles` );
```

Comment:
Is this a good idea? 🤔 
Can't we render the modal here directly instead of an expensive round-trip to the site editor?

Reply:
I think you're right. I somehow interpreted the "Clicking on this step triggers the same modal we use in the editor (see attached) with adjusted copy" requirement from https://github.com/Automattic/dotcom-forge/issues/1057#issuecomment-1290130716 as a redirect to the site editor because of the screenshot, but the screenshot was only provided to give some context about the modal itself, not because we need to redirect to the site editor. I'll try to show the modal here and avoid the redirect.

Reply:
Having the modal on the same page makes more sense to me too.

Reply:
+1

Reply:
If it's too much of a mess to move the modal into a shared location, IMHO it's simple enough that we could really just rewrite here, also because the copy is different anyway.

Reply:
https://github.com/Automattic/wp-calypso/pull/69487/commits/7834f7bba101cd591e66a837d5474b7a3ba7fb89 and https://github.com/Automattic/wp-calypso/pull/69487/commits/104591c1a8c110db9e382a55aaea980bb65ef68b